cli/flox/src/commands/publish.rs

```diff
@@ -101,21 +110,47 @@ impl Publish {
         let build_metadata =
             check_build_metadata(&flox, &env_metadata, &path_env, &FloxBuildMk, &package)?;
 
-        let cache = if no_store {
+        let cache = if metadata_only {
             None
         } else {
-            merge_cache_options(config.flox.publish, cache_args)?
+            // Get the ingress URI for this catalog if it has one configured.
+            let ingress_uri = flox.catalog_client.get_ingress_uri(&catalog_name).await?;
+            let catalog_has_ingress_uri = ingress_uri.is_some();
```

Comment:
followup suggestion: seems like we should test some of the error cases

Reply:
100%, I expect there will be lots of followup PRs for all of the stuff that got merged over the last few days.